packages/block-editor/src/components/dimension-control/index.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import { isFunction } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	Icon,
+	SelectControl,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+import { withInstanceId } from '@wordpress/compose';
+
+import {
+	Fragment,
+} from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import sizesTable, { findSizeBySlug } from './sizes';
+
+export function DimensionControl( props ) {
+	const { label, icon, iconLabel = 'all', sizes = sizesTable, currentSize, onSpacingChange, onReset, className = '' } = props;
+
+	/**
+	 * Determines the size from the size slug (eg: `medium`)
+	 * and decides whether to call the change or reset callback
+	 * handlers
+	 * @param  {string} val the DOMEvent event.target
+	 * @return {void}
+	 */
+	const onChangeSpacingSize = ( val ) => {
+		const theSize = findSizeBySlug( sizes, val );
+
+		if ( ! theSize || currentSize === theSize.slug ) {
+			resetSpacing();
+		} else if ( isFunction( onSpacingChange ) ) {
+			onSpacingChange( theSize.slug );
+		}
+	};
+
+	/**
+	 * Applies the callback to handle resetting
+	 * a dimension spacing values
+	 * @return {void}
+	 */
+	const resetSpacing = () => {
+		if ( isFunction( onReset ) ) {
+			onReset();
+		}
+	};
+
+	/**
+	 * Converts the sizes lookup tablet
+	 * to a format suitable for use in the
+	 * <SelectControl /> options prop
+	 * @param  {Array} theSizes the array of sizes objects
+	 * @return {Array}       the array of options with default option prepended
```

Comment:
Please adjust the formatting of docblocks to be consistent with Gutenberg practices (see also the previous docblocks)

Reply:
Honestly, I couldn't easily locate the best practices which says something in and of itself. I've reached out to Riad to ask whether there's tooling in place to auto-format these because relying on humans to do this sort of thing might not be a great way forward.

Reply:
Here's the WordPress ones:

JavaScript Documentation Standards
https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/

The Gutenberg ones slightly differ for minor details, not sure they're documented anywhere.